packages/core/src/lib/modal.svelte

```diff
@@ -73,15 +75,29 @@ $: if (typeof document !== 'undefined') {
   document.body.classList.toggle('overflow-hidden', isOpen);
 }
 
-$: headingElement?.focus();
+// Focus Logic
+// If one slot is passed focus slot element
+// if two slots are passed reference focusPrimaryElement and focus primary or secondary slot
+$: {
+  if (isOpen) {
+    const slotValues = slotElement?.querySelectorAll('button');
+    if (slotValues?.length !== undefined && slotValues.length > 0) {
+      if (focusPrimaryElement) {
+        slotValues.item(1).focus();
+      } else {
+        slotValues.item(0).focus();
+      }
+    }
+  }
+}
 </script>
 
 <svelte:window on:keydown={isOpen ? handleEscapePress : undefined} />
 
 {#if isOpen}
   <div
     class="fixed left-0 top-0 z-50 flex h-full w-full items-center justify-center bg-black bg-opacity-40"
-    role="dialog"
+    role="alertdialog"
```

Comment:
Should we make this `role` configurable? I think defaulting to `alertdialog` is a bit aggressive; it feels safer to default to `dialog` but allow a called to pass in `role="alertdialog"`

Reply:
Yupp good call, added a prop